src/middlewared/middlewared/plugins/disk_/configure_swap.py

```diff
@@ -0,0 +1,223 @@
+import os
+import platform
+import subprocess
+
+from collections import defaultdict
+
+from middlewared.service import CallError, private, Service
+from middlewared.utils import Popen, run
+
+IS_LINUX = platform.system().lower() == 'linux'
+MIRROR_MAX = 5
+
+
+class DiskService(Service):
+
+    @private
+    async def swaps_configure(self):
+        """
+        Configures swap partitions in the system.
+        We try to mirror all available swap partitions to avoid a system
+        crash in case one of them dies.
+        """
+        used_partitions_in_mirror = set()
+        create_swap_devices = {}
+        disks = [i async for i in await self.middleware.call('pool.get_disks')]
+        disks.extend(await self.middleware.call('boot.get_disks'))
+        existing_swap_devices = {'mirrors': [], 'partitions': []}
+        mirrors = await self.middleware.call('disk.get_mirrors')
+        encrypted_mirrors = {m['encrypted_provider']: m for m in mirrors if m['encrypted_provider']}
+        all_partitions = {p['name']: p for p in await self.middleware.call('disk.list_all_partitions')}
+
+        for device in await self.middleware.call('disk.get_swap_devices'):
+            if device in encrypted_mirrors or device.startswith(('/dev/md', 'mirror/')):
+                # This is going to be complete path for linux and mirror/swapname.eli for freebsd
```

Comment:
Why for linux is absolute path and for freebsd it returns relative? Thats a bit confusing, no?